tpcc/main.go

```diff
@@ -180,14 +181,17 @@ func main() {
 		}
 	} else {
 		// 10 workers per warehouse.
-		workers = make([]*worker, *warehouses*10)
-		for i := 0; i < *warehouses; i++ {
-			for j := 0; j < 10; j++ {
-				idx := i*10 + j
-				workers[idx] = newWorker(idx, db, &wg)
-				go workers[idx].run(errCh, i)
+		go func() {
+			workers = make([]*worker, 0, *warehouses*10)
+			for i := 0; i < 10; i++ {
+				for j := 0; j < *warehouses; j++ {
+					idx := i**warehouses + j
+					workers = append(workers, newWorker(idx, db, &wg))
+					time.Sleep(time.Duration(*rampMs/uint64(cap(workers))) * time.Millisecond)
```

Comment:
Is this so all of the workers don't start at the same time? Perhaps it would be better for the workers to start with a randomized think time.